frontend/app-development/hooks/mutations/useDeleteDataModelMutation.ts

```diff
@@ -32,7 +32,7 @@ export const useDeleteDataModelMutation = () => {
       queryClient.removeQueries({
         queryKey: [QueryKey.JsonSchema, org, app, xsdPath],
       });
-      queryClient.invalidateQueries({ queryKey: [QueryKey.AppMetadataModelIds, org, app] });
+      queryClient.removeQueries({ queryKey: [QueryKey.AppMetadataModelIds, org, app] });
```

Comment:
Ah yes, that is a good question.
The reason for this change is related to testing the validation for the data model selector. When a selected data model no longer existed, an unnecessary/confusing toaster would appear, despite the functionality working as intended. The validation should cover this instead. See the video below for how it behaved before the change.

The toaster is triggered in `useValidDataModels.ts`, where it still uses old data, specifically here:
```
  const { data: dataModels, isPending: dataModelsIsPending } = useAppMetadataModelIdsQuery(
    org,
    app,
    false,
  );
  ```
With invalidation, the toaster kept showing up, while removeQueries would fix that. I discussed this with @framitdavid , who explained that invalidateQuery triggers a background fetch, causing the hook to validate old data until new data arrives. By using removeQuery, the cache is cleared completely, preventing the hook from validating old data and ensuring it waits for new data before running validation.

I haven't discovered any bugs after making this change. 🤔 

https://github.com/user-attachments/assets/15b95538-cb12-4e63-b46a-373a6f430e82

Reply:
Yes, how this code is implemented today if we do `invalidateQueries` it would send a background request to validate against a deleted model since the model exists in the cache. By removing the the queries (cache), it would not trigger a background fetch with the old model. 😊

Reply:
> it would send a background request to validate against a deleted model since the model exists in the cache
But should the model exist in the cache if it is deleted? Isn't that why we have invalidated the query when we delete a model - to ensure the cache is up to date? 🤔 
Might be that I misunderstand something here tho.

Reply:
@standeren, yes we invalidate queries to ensure the cache is updated. Maybe the `useValidDataModels.ts` is re-rendered before the response with the new cache is finished and therefore it's validates against a deleted data model. 

@lassopicasso have you tried to add `isRefetching` check instead of just `isPending`? I think  the `isPending` is `true` only the first time, when the query does not have any data.  By removing the queries/cache, the `isPending` would be `true`.

```typescript 
 const { data: dataModels, isPending: dataModelsIsPending, isRefetching: isRefetchingDataModels } = useAppMetadataModelIdsQuery(
    org,
    app,
    false,
  );
```

Reply:
You mean, use both `isPending` and `isRefetching` to wait for fetching `useDataModelMetadataQuery`. 
And at the same time change ivalidateQueries instead of removeQueries?

Reply:
Yes, use both `isPending` and `isRefetching`. 😊

Reply:
Correct it is the first query, tested this and the toaster still shows up :/ 
It is tempting to keep removeQueries for now as I don't see any unexpected behaviors.

Reply:
Alrighty, adding `&&` instead of `| |` solved the case! Thx @framitdavid and @standeren for pointing this out that resulted in a better solution!